diff --git a/src/bootstrap/step.rs b/src/bootstrap/step.rs
index a5c0d11d21..ff34497613 100644
--- a/src/bootstrap/step.rs
+++ b/src/bootstrap/step.rs
@@ -667,7 +667,7 @@ pub fn build_rules<'a>(build: &'a Build) -> Rules {
          .only_host_build(true)
          .run(move |s| dist::std(build, &s.compiler(), s.target));
     rules.dist("dist-mingw", "path/to/nowhere")
-         .default(true)
+         .default(false)
          .only_host_build(true)
          .run(move |s| {
              if s.target.contains("pc-windows-gnu") {
@@ -675,7 +675,7 @@ pub fn build_rules<'a>(build: &'a Build) -> Rules {
              }
          });
     rules.dist("dist-src", "src")
-         .default(true)
+         .default(false)
          .host(true)
          .only_build(true)
          .only_host_build(true)
diff --git a/src/librustc_llvm/ffi.rs b/src/librustc_llvm/ffi.rs
index 26c7a9166e..ba6b4a42f3 100644
--- a/src/librustc_llvm/ffi.rs
+++ b/src/librustc_llvm/ffi.rs
@@ -1668,4 +1668,7 @@ extern "C" {
 // during llvm-config?
 #[cfg(windows)]
 #[link(name = "ole32")]
+#[link(name = "ffi")]
+#[link(name = "gcc_eh")]
+#[link(name = "pthread")]
 extern "C" {}
diff --git a/src/libunwind/build.rs b/src/libunwind/build.rs
index ed3d5212bf..7dd53e2026 100644
--- a/src/libunwind/build.rs
+++ b/src/libunwind/build.rs
@@ -36,6 +36,7 @@ fn main() {
         println!("cargo:rustc-link-lib=gcc_pic");
     } else if target.contains("windows-gnu") {
         println!("cargo:rustc-link-lib=gcc_eh");
+        println!("cargo:rustc-link-lib=pthread");
     } else if target.contains("fuchsia") {
         println!("cargo:rustc-link-lib=unwind");
     }
