diff --git a/src/bootstrap/step.rs b/src/bootstrap/step.rs
index 6008fa81c6..1f666e0a29 100644
--- a/src/bootstrap/step.rs
+++ b/src/bootstrap/step.rs
@@ -711,7 +711,7 @@ pub fn build_rules<'a>(build: &'a Build) -> Rules {
          .only_host_build(true)
          .run(move |s| dist::std(build, &s.compiler(), s.target));
     rules.dist("dist-mingw", "path/to/nowhere")
-         .default(true)
+         .default(false)
          .only_host_build(true)
          .run(move |s| {
              if s.target.contains("pc-windows-gnu") {
@@ -719,7 +719,7 @@ pub fn build_rules<'a>(build: &'a Build) -> Rules {
              }
          });
     rules.dist("dist-src", "src")
-         .default(true)
+         .default(false)
          .host(true)
          .only_build(true)
          .only_host_build(true)
diff --git a/src/librustc_llvm/build.rs b/src/librustc_llvm/build.rs
index a8def4bafd..7b389fecfd 100644
--- a/src/librustc_llvm/build.rs
+++ b/src/librustc_llvm/build.rs
@@ -267,5 +267,8 @@ fn main() {
     // during llvm-config?
     if target.contains("windows") {
         println!("cargo:rustc-link-lib=ole32");
+        println!("cargo:rustc-link-lib=ffi");
+        println!("cargo:rustc-link-lib=gcc_eh");
+        println!("cargo:rustc-link-lib=pthread");
     }
 }
