diff --git a/install-template.sh b/install-template.sh
index 692a394..4df25e1 100644
--- a/install-template.sh
+++ b/install-template.sh
@@ -623,6 +623,11 @@
 		local _f="$(echo "$_file" | sed 's/^share\/man\///')"
 		_file_install_path="$CFG_MANDIR/$_f"
 	    fi
+      if echo "$_file" | grep "^share/doc/" > /dev/null
+      then
+          local _f="$(echo "$_file" | sed 's/^share\/doc\/rust\///')"
+          _file_install_path="$CFG_DOCDIR/$_f"
+      fi
 
 	    # Make sure there's a directory for it
 	    make_dir_recursive "$(dirname "$_file_install_path")"
@@ -834,6 +839,7 @@ valopt components "" "comma-separated list of components to install"
 flag list-components "list available components"
 valopt libdir "$CFG_DESTDIR_PREFIX/lib" "install libraries"
 valopt mandir "$CFG_DESTDIR_PREFIX/share/man" "install man pages in PATH"
+valopt docdir "$CFG_DESTDIR_PREFIX/share/doc/rust" "install documentation in PATH"
 opt ldconfig 1 "run ldconfig after installation (Linux only)"
 opt verify 1 "obsolete"
 flag verbose "run with verbose output"
