diff --git a/src/bootstrap/step.rs b/src/bootstrap/step.rs
index ee5b61062f..819749f7c6 100644
--- a/src/bootstrap/step.rs
+++ b/src/bootstrap/step.rs
@@ -626,7 +626,7 @@ pub fn build_rules<'a>(build: &'a Build) -> Rules {
          .only_host_build(true)
          .run(move |s| dist::std(build, &s.compiler(), s.target));
     rules.dist("dist-mingw", "path/to/nowhere")
-         .default(true)
+         .default(false)
          .only_host_build(true)
          .run(move |s| {
              if s.target.contains("pc-windows-gnu") {
diff --git a/src/librustc_llvm/ffi.rs b/src/librustc_llvm/ffi.rs
index 8a958cf1ac..da9419cf9a 100644
--- a/src/librustc_llvm/ffi.rs
+++ b/src/librustc_llvm/ffi.rs
@@ -1667,4 +1667,7 @@ extern "C" {
 // during llvm-config?
 #[cfg(windows)]
 #[link(name = "ole32")]
+#[link(name = "ffi")]
+#[link(name = "gcc_eh")]
+#[link(name = "pthread")]
 extern "C" {}
diff --git a/src/libunwind/build.rs b/src/libunwind/build.rs
index f18b694d3d..1de4b71164 100644
--- a/src/libunwind/build.rs
+++ b/src/libunwind/build.rs
@@ -33,6 +33,7 @@ fn main() {
         println!("cargo:rustc-link-lib=gcc_pic");
     } else if target.contains("windows-gnu") {
         println!("cargo:rustc-link-lib=gcc_eh");
+        println!("cargo:rustc-link-lib=pthread");
     } else if target.contains("fuchsia") {
         println!("cargo:rustc-link-lib=unwind");
     }
