diff --git a/install-template.sh b/install-template.sh
index 02070c3..6c8ba39 100644
--- a/install-template.sh
+++ b/install-template.sh
@@ -601,6 +601,8 @@ install_components() {
 	append_to_file "$_component" "$_md/components"
 	critical_need_ok "failed to update components list for $_component"
 
+    # Replace \ with /
+	local _sed=`sed -i 's|\\\\|/|g' $_input_manifest`
 	# Now install, iterate through the new manifest and copy files
 	local _directive
 	while read _directive; do
